sdk/src/main/java/ly/count/android/sdk/ModuleRequestQueue.java

```diff
@@ -217,6 +231,32 @@ synchronized public void requestQueueEraseAppKeysRequestsInternal() {
         attemptToSendStoredRequestsInternal();
     }
 
+    /**
+     * Send request data after removing the predefined keys
+     */
+    synchronized public Countly addDirectRequestInternal(Map<String, String> requestMap) {
+        L.i("[Countly] Calling addDirectRequest");
+        if (!_cly.isInitialized()) {
+            throw new IllegalStateException("Countly.sharedInstance().init must be called before adding direct request");
+        }
+
+        if (!consentProvider.getConsent(Countly.CountlyFeatureNames.events)) {
+            return _cly;
+        }
+
+        if (requestMap == null) {
+            L.d("[ModuleRequestQueue] addDirectRequest, provided requestMap was null, returning");
+
+            return _cly;
+        }
+
+        requestMap.keySet().removeAll(Arrays.asList(preDefinedKeys));
```

Comment:
Compare how many keys were removed and print a warning if any were removed. Print also the count.

Reply:
And if resulting count is 0, print appropriate warning and return